src/validation.cpp

```diff
@@ -2914,9 +2914,9 @@ bool CChainState::ResetBlockFailureFlags(CBlockIndex *pindex) {
         }
         pindex = pindex->pprev;
     }
-    return true;
 }
-bool ResetBlockFailureFlags(CBlockIndex *pindex) {
+
+void ResetBlockFailureFlags(CBlockIndex *pindex) {
     return g_chainstate.ResetBlockFailureFlags(pindex);
```

Comment:
`return` in a `void` context

Reply:
We want to assert that the return code type is the same for both functions, so this looks correct as is.

Reply:
So this is pending future use? If not I don't follow.

Reply:
Imagine that `Chainstate::ResetBlockFailureFlags` returns `bool` in the future. This prevents the code from compiling, whereas removing the return would happily compile and hide a potential bug, no?

Also, changing it to something else in the future keeps the future diff smaller while keeping the return.